frame/support/src/lib.rs

```diff
@@ -235,7 +244,45 @@ macro_rules! parameter_types {
 				I::from(Self::get())
 			}
 		}
-	}
+	};
+	(
+		$(
+			$( #[ $attr:meta ] )*
+			$vis:vis static $name:ident: $type:ty = $value:expr;
+		)*
+	) => (
+		#[cfg(no_std)]
```

Comment:
Here you should use `cfg(not(feature = "std"))`.

Also we should do a trick ;)

Create an extra macro that is called by this match arm here. This macro will contain this and the next match arm when the `std` feature is enabled. You create the same macro a second time, this time for `no_std`. This will just contain the `compile_error!`. Doing it this way will not require the crate where we use this macro to have the `std` feature!

Reply:
> Doing it this way will not require the crate where we use this macro to have the std feature!

I think exactly the way to allow this to be `std`-independent is to use `#[cfg(no_std)]`. What you described binds us to an `std` feature existing. 

Either way, I interpret your suggestion as: 

```
#[cfg(not(feature = "std"))]
macro_rules! parameter_types_impl_thread_local {
	( $( $any:tt )* ) => {
		compile_error!("static parameter types is only available in std and for testing.");
	};
}

#[cfg(feature = "std")]
macro_rules! parameter_types_impl_thread_local {
	(
		$(
			$( #[ $attr:meta ] )*
			$vis:vis static $name:ident: $type:ty = $value:expr;
		)*
	) => {
            // stuff
        } 
}
```

Is this what you mean or have I misunderstood it?

Reply:
Exactly 👍

Reply:
updated, I was expecting though that the new macro need not to be `#[macro_export]` ed and could have been used internally, but that didn't work and I still had to export + use them with `$crate`.